db/compaction_job.cc

```diff
@@ -521,7 +521,13 @@ void CompactionJob::GenSubcompactionBoundaries() {
     }
 
     const Slice b = *it;
+
+    // ApproximateSize could potentially create table reader iterator to seek
+    // to the index block and may incur I/O cost in the process. Unlock db
+    // mutex to reduce contention
+    db_mutex_->Unlock();
```

Comment:
do you mind explaining the reasoning behind why this solution is safe? To me it is not clear whether `v` is pinned, for example.